libs/stats/odc/stats/_fc_percentiles.py

```diff
@@ -0,0 +1,81 @@
+"""
+Fractional Cover Percentiles
+"""
+from typing import Optional, Tuple
+from itertools import product
+import xarray as xr
+from odc.stats.model import Task
+from odc.algo.io import load_with_native_transform
+from odc.algo import keep_good_only
+from odc.algo._percentile import xr_percentile
+from odc.algo._masking import _or_fuser
+from .model import StatsPluginInterface
+from . import _plugins
+
+
+class StatsFCP(StatsPluginInterface):
+    
+    NAME = "ga_fc_percentiles"
+    SHORT_NAME = NAME
+    VERSION = "0.0.1"
+    PRODUCT_FAMILY = "fc_percentiles"
+
+    def __init__(
+        self,
+        resampling: str = "bilinear",
+    ):
+        self.resampling = resampling
+
+    @property
+    def measurements(self) -> Tuple[str, ...]:
+        _measurments = [f"{b}_pc_{p}" for b, p in product(["pv", "bs", "npv", "ue"], ["10", "50", "90"])]
+        _measurments.append("is_ever_wet")
+        return _measurments
+
+    @staticmethod
+    def _native_tr(xx):
+        """
+        Loads in the data in the native projection. It performs the following:
+
+        1. Loads all the fc and WOfS bands
+        2. Extracts the clear dry and clear wet flags from WOfS
+        3. Drops the WOfS band
+        4. Stores the clear wet flags for QA later on 
+        5. Masks out all pixels that are not clear and dry to a nodata value of 255
+        6. Discards the clear dry flags
+        """
+        # use the dry flag to indicate good measurementd
+        dry = xx.water == 0
+
+        # keep the clear wet measurements to calculate the QA band in reduce
+        xx["wet"] = xx.water == 128
+        xx = xx.drop_vars(["water"])
+        return keep_good_only(xx, dry, nodata=255)
+
+    def input_data(self, task: Task) -> xr.Dataset:
+
+        chunks = {"y": -1, "x": -1}
+
+        xx = load_with_native_transform(
+            task.datasets,
+            bands=["water", "pv", "bs", "npv", "ue"],
+            geobox=task.geobox,
+            native_transform=self._native_tr,
+            fuser=None,
+            groupby=None,
```

Comment:
grouping by solar day and fusing is required, otherwise you end up with double counting observations in North-South overlap.

Reply:
I have a question re. the `_nodata_fuser`: does it consider the bands seperately?

It looks like it does, and if so I think something like this could happen:

(two copies of the same pixel on the same day)
```
pv    255    80
dry   True    False
```

Gets fused to:

```
pv    80
dry    True
```

causing us to incorrectly use a wet measurement. If that makes sense?

Reply:
custom fuser is given a dataset object filled with dask arrays, it's up to you which bands to fuse with which fuser, so you want to use _nodata_fuser for data bands and `and` or `or`  for boolean bands.
keep in mind that fusing happens several times: once for every native projection and then again after reprojection.

Reply:
oh wait I just realised that the `fuser` is called after the `native_transform` so my comment above isn't an issue. All the measurements that aren't dry and clear will be set to `nodata` before the fuser anyway so I can just use the `choose_first_valid` for the `fc` bands.

And I think using `or` for the `wet` band is the right choice here.

How do I apply different fusers to different bands?

Reply:
fuser is applied in native projection as well as in final projection

Reply:
but yes, after native transform was called

Reply:
custom fuser is given an `xr.Dataset` object that was returned from native_transform, you can slice into it with data band names, apply nodata fuser to those, then add masking bands that are fused with `or` afterwards.

Reply:
oh I think I've got it - I just run load multiple times